integration_tests/src/main/python/hybrid_parquet_test.py

```diff
@@ -290,6 +305,478 @@ def test_hybrid_parquet_filter_pushdown_timestamp(spark_tmp_path):
+    # Date:
+    # LastDay, WeekOfYear,DayOfMonth, DayOfWeek, DayOfYear, Month, Quarter, Second, UnixMicros, UnixMillis, UnixSeconds, WeekDay, Year
+    "(last_day(date1) == date1)",
+    "(weekofyear(date1) == 1)",
+    "(dayofmonth(date1) == 1)",
+    "(dayofweek(date1) == 1)",
+    "(dayofyear(date1) == 1)",
+    "(month(date1) == 1)",
+    # Date, Int
+    # AddMonths
+    "(add_months(date1, 1) == date1)",
+    # Date, Date:
+    # DateDiff
+    "(date_diff(date1, date2) == 1)",
+    # unit, Int, Date:
+    # DateAdd,DateSub
+    "(date_add(date1, 1) == date2)",
+    "(date_sub(date1, 1) == date2)",
+    # Array[Int], lambda:
+    # ArrayExists,ArrayForAll
+    "(exists(array_int_1, x -> x == 1))",
+    "(forall(array_int_1, x -> x == 1))",
+    # Array[Int], Int:
+    # ArrayContains,ArrayPosition,ArrayRemove,ElementAt
+    "(array_contains(array_int_1, 1))",
+    "(array_position(array_int_1, 1) == 1)",
+    "(array_remove(array_int_1, 1) == array_int_1)",
+    "(element_at(array_int_1, 1) == 1)",
+    # Array[Int]:
+    # ArrayDistinct,ArrayMax,ArrayMin,ArraySort,Shuffle,Size
+    "(array_distinct(array_int_1) == array_int_1)",
+    "(array_max(array_int_1) == 1)",
+    "(array_min(array_int_1) == 1)",
+    "(sort_array(array_int_1) == array_int_1)",
+    pytest.param("(shuffle(array_int_1) == array_int_1)", marks=pytest.mark.xfail(reason='result not correct')),
+    "(size(array_int_1) == 1)",
+    # Array[Int], Array[Int]:
+    # ArrayExcept,ArrayIntersect,ArraysZip
+    "(array_except(array_int_1, array_int_2) == array_int_1)",
+    "(array_intersect(array_int_1, array_int_2) == array_int_1)",
+    "(arrays_zip(array_int_1, array_int_2) == " + 
+    "array(struct(1 as array_int_1, 1 as array_int_2), struct(2 as array_int_1, 2 as array_int_2), struct(3 as array_int_1, 3 as array_int_2)))",
+    # Array[Array[Int]]:
+    # Flatten
+    "(flatten(array_array_int_1) == array_int_1)",
+    # Array[Int], Array[Int], lambda:
+    # ZipWith
+    "(zip_with(array_int_1, array_int_2, (x, y) -> x + y) == array(2,4,6))",
+    # String:
+    # Ascii,BitLength,Hex,Length,Lower,Reverse,Sha1,SoundEx,StringTrim,StringTrimLeft,StringTrimRight,Upper
+    "(ascii(str1) == 79)",
+    "(bit_length(str1) == 10)",
+    "(hex(str1) == '4f')",
+    "(length(str1) == 5)",
+    "(lower(str1) == 'hello')",
+    "(reverse(str1) == 'olleh')",
+    pytest.param("(sha1(str1) == 'b10a8db164e0754105b7a99be72e3fe5')", marks=pytest.mark.xfail(reason='binary type not supported yet')),
+    "(soundex(str1) == 'h400')",
+    "(trim(str1) == 'hello')",
+    "(ltrim(str1) == 'hello')",
+    "(rtrim(str1) == 'hello')",
+    "(upper(str1) == 'HELLO')",
+    # Int, Int:
+    # ArrayRepeat,BitwiseAnd,BitwiseOr,BitwiseXor,EqualNullSafe,EqualTo,GreaterThan,GreaterThanOrEqual,
+    # Greatest,Least,LessThan,LessThanOrEqual,Remainder,ShiftLeft,ShiftRight
+    "(array_repeat(array_int_1, 2) == array(array_int_1, array_int_1))",
+    "(int1 & int2 == 1)",
```

Comment:
So the only values that would make this match are `0 & 1`, `1 & 1`, `1 & 0` out of all possible integer values randomly selected, twice? 2^32 * 2 ^ 32. I don't think this is a good test. But I am not sure how to do a better one unless we explicitly reduce the window of possibility.

Reply:
Updated, and I reduced the window of possibility in datagen for this one and many other cases to make sure we can filter / filter out at least 1 items when scan.